extensions/wikia/ArticleVideo/ArticleVideo.hooks.php

```diff
@@ -5,9 +5,26 @@ public static function onBeforePageDisplay( \OutputPage $out, \Skin $skin ) {
 		$wg = F::app()->wg;
 		$title = $wg->Title->getPrefixedDBkey();
 
-		if ( isset( $wg->articleVideoFeaturedVideos[$title] ) ) {
-			\Wikia::addAssetsToOutput( 'article_video_scss' );
-			\Wikia::addAssetsToOutput( 'article_video_js' );
+		if ( isset( $wg->articleVideoRelatedVideos ) ) {
+			$relatedVideo = ArticleVideoController::getRelatedVideoData( $wg->articleVideoRelatedVideos, $title );
+		}
+
+		if ( ( $wg->enableArticleFeaturedVideo &&
+		       isset( $wg->articleVideoFeaturedVideos[$title] ) ) ||
+		     ( $wg->enableArticleRelatedVideo && isset( $relatedVideo ) )
+		) {
+			\Wikia::addAssetsToOutput( 'ooyala_scss' );
+			\Wikia::addAssetsToOutput( 'ooyala_js' );
+		}
+
+		if ( $wg->enableArticleFeaturedVideo && isset( $wg->articleVideoFeaturedVideos[$title] ) ) {
+			\Wikia::addAssetsToOutput( 'article_featured_video_scss' );
+			\Wikia::addAssetsToOutput( 'article_featured_video_js' );
+		}
+		
+		if ( $wg->enableArticleRelatedVideo && isset( $relatedVideo ) ) {
```

Comment:
We check this condition 3 times in this file and then again in the controller. How about extracting it to a method? The same goes for the featured video.